src/main/telemetry/srxl.c

```diff
@@ -155,17 +159,54 @@ typedef struct
 
 #define SPEKTRUM_RPM_UNUSED 0xffff
 #define SPEKTRUM_TEMP_UNUSED 0x7fff
+#define MICROSEC_PER_MINUTE 60000000
 
-bool srxlFrameRpm(sbuf_t *dst, timeUs_t currentTimeUs)
+//Original range of 1 - 65534 uSec gives an RPM range of 915 - 60000000rpm, 60MegaRPM
+#define SPEKTRUM_MIN_RPM 999      // Min RPM to show the usre, indicating RPM is really below 999
+#define SPEKTRUM_MAX_RPM 60000000
+
+uint16_t getMotorAveragePeriod(void)
 {
+
+#if defined( USE_ESC_SENSOR_TELEMETRY) || defined( USE_DSHOT_TELEMETRY)
+    uint32_t rpm = 0;
     uint16_t period_us = SPEKTRUM_RPM_UNUSED;
-#ifdef USE_ESC_SENSOR_TELEMETRY
+
+#if defined( USE_ESC_SENSOR_TELEMETRY)
     escSensorData_t *escData = getEscSensorData(ESC_SENSOR_COMBINED);
-    if (escData != NULL && escData->rpm > 0) {
-        period_us = 60000000 / escData->rpm; // revs/minute -> microSeconds
+    if (escData != NULL) {
+        rpm = escData->rpm;
     }
 #endif
 
+#if defined( USE_DSHOT_TELEMETRY)
+    if (useDshotTelemetry) {
+        uint16_t motors = getMotorCount();
+
+        if (motors > 0) {
+            for (int  motor = 0; motor < motors; motor++) {
+                rpm += getDshotTelemetry(motor);
```

Comment:
`getDshotTelemetry()` returns the ERPM frequency, so in order for it to have RPM dimension it needs to be converted, see: https://github.com/betaflight/betaflight/blob/9e715845c3c0214e1a25e4ed473fa532b6313557/src/main/osd/osd_elements.c#L165-L175

Reply:
I saw that code, and I did not understand it. I looked at the cli status code instead. And the values reported over Spektrum TM matches those presented by cli. And cli status says rpm, not erpm. 
But OK, I'll give it that expression a try.

Reply:
I had a quick look through `src/main/sensors/rpm_filter.c`, and cobbling together how `getDshotTelemetry()` is handled (which is verified by RPM filtering working) the above expression seems to be right:

```
#define SECONDS_PER_MINUTE      60.0f
#define ERPM_PER_LSB            100.0f

erpmToHz = ERPM_PER_LSB / SECONDS_PER_MINUTE  / (motorConfig()->motorPoleCount / 2.0f);

filteredMotorErpm[motor] = pt1FilterApply(&rpmFilters[motor], getDshotTelemetry(motor));

motorFrequency[motor] = erpmToHz * filteredMotorErpm[motor];
```

I want to look into refactoring some of the bidirectional Dshot code and add some nice getters for things like RPM once 4.0 is out.